articles/implementing-search-view-in-android/index.md

```diff
@@ -0,0 +1,339 @@
+
+```kotlin
+data class Person(
+    val name: String,
+    val age: Int
+)
+```
+
+### G). Setting up RecyclerView Adapter
+A RecyclerView is a flexible widget that can bind a variety of data with respect to its type. However, there's no specific type of data that should be provided to it. An adapter helps us prepare the RecyclerView to handle the given data. 
+
+Inside the adapter package (created at step D(i)), create a Kotlin class named `PersonAdapter`. This class will extend a RecyclerView Adapter and accept an `ArrayList` of type `Person` (the data class created in the previous step). Unlike a List, an ArrayList supports the addition and deletion of elements at runtime which is why we're using it.
+
+Copy and paste the code below in the `PersonAdapter.kt` file
+
+```kotlin
+class PersonAdapter(
+    var list: ArrayList<Person>
+) : RecyclerView.Adapter<PersonAdapter.ViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+
+        return ViewHolder(
+            RowItemBinding.inflate(
+                LayoutInflater.from(parent.context),
+                parent,
+                false
+            )
+        )
+    }
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(
+            list[position]
+        )
+    }
+
+    override fun getItemCount(): Int = list.size
+
+    inner class ViewHolder(private var item: RowItemBinding) : RecyclerView.ViewHolder(item.root) {
+        fun bind(person: Person) {
+            item.name.text = person.name.trim()
+            item.age.text = person.age.toString().trim()
+        }
+    }
+}
+```
+
+Press `Alt`+`Enter` to fix missing imports. The above code enables the RecyclerView to bind views and populate them with data in the order it appears in the ArrayList. [This tutorial](https://developer.android.com/reference/android/widget/Adapter) elaborates further on how an Adapter works.
+
+### H). How Linear Search Algorithm works
+Just like any other task, searching is a process that requires a series of steps to be performed. In this tutorial, we'll make use of the `Linear search algorithm`. 
+
+First set up two lists, one containing all the data under consideration, and leave the other one empty. We'll therefore loop through the elements in the first list comparing each content with the given keyword. In this case, the keyword is the `text` typed by the user in the SearchView. If a match is found, the element containing the match is cloned into the second list and the RecyclerView is updated with the new list. 
+
+When working with a very large list or a list of unknown size, it is recommended to update the RecyclerView after looping is done. Otherwise, the app might land into an `ANR` (App Not Responding) situation due to high memory consumption. Also, the newly created list should be cleared when not required for the same reason.
```

Comment:
Could you please elaborate this point. I think waiting for the loop to finish should result to ANR since a process is running but the UI is not being updated.